app/controllers/incidents_controller.rb

```diff
@@ -166,6 +166,7 @@ def update
   def incident_params
     data = params.require(:incident).permit!
     sup_report_attrs = data[:supervisor_incident_report_attributes]
+    @incident.driver_incident_report = IncidentReport.create( user_id: current_user.id ) if current_user.driver?
```

Comment:
I feel uncomfortable creating an object in the params. If not for any reason other than the question: What if it doesn't work? There's no error handling here, that's done (as is the "Rails Way") in the create action.
There are a couple standard or at least more future-proof ways of handling the problem of "create a driver incident report also if the current user is a driver". 
Since `Incident` `accepts_nested_attributes_for` `IncidentReport`, you could just handle this in the same way the supervisor incident report is, with `incident_driver_incident_report_attributes`. See the way `sup_report_attrs` is created and used.
What you want to end up with is `data` that can be used to create an object, not some data and also an already-created object.